plugins/system_upgrade.py

```diff
@@ -568,6 +577,9 @@ def run_reboot(self):
         if not self.opts.tid[0] == "reboot":
             return
 
+        if self.opts.poweroff_after is not None:
```

Comment:
We could probably move it inside the `transaction_download` method aside the other states setters.